benches/enrichment_tables_file.rs

```diff
@@ -0,0 +1,123 @@
+use criterion::{criterion_group, criterion_main, BatchSize, Criterion};
+use std::collections::BTreeMap;
+use vector::enrichment_tables::{file::File, Condition, Table};
+
+criterion_group!(
+    name = benches;
+    config = Criterion::default().noise_threshold(0.02).sample_size(10);
+    targets = benchmark_enrichment_tables_file
+);
+criterion_main!(benches);
+
+fn benchmark_enrichment_tables_file(c: &mut Criterion) {
```

Comment:
Nit: this could probably be refactored to have a list of cases at the top to reduce some of the duplication. Not a big deal, but just a thought for the future.